tests/generation/test_utils.py

```diff
@@ -2805,6 +2805,27 @@ def test_beam_search_low_memory(self):
         )
         self.assertListEqual(low_output.tolist(), high_output.tolist())
 
+    @slow
+    def test_watermark_generation(self):
+        tokenizer = GPT2Tokenizer.from_pretrained("openai-community/gpt2")
+        model = AutoModelForCausalLM.from_pretrained("openai-community/gpt2").to(torch_device)
+        tokenizer.pad_token_id = tokenizer.eos_token_id
+        model_inputs = tokenizer("I will be", return_tensors="pt").to(torch_device)
+
+        output = model.generate(**model_inputs, watermark=True, tokenizer=tokenizer, do_sample=False, max_length=10)
+        output_selfhash = model.generate(
+            **model_inputs,
+            watermark=True,
+            tokenizer=tokenizer,
+            seeding_scheme="selfhash",
+            do_sample=False,
+            max_length=10,
+        )
+
+        # as long as we use the same inputs, hashing key and device, we can be sure that output are deterministic when greedy decoding
```

Comment:
Since watermarking is not very deterministic, I added this kind of test. Another option is to use the actual detector and detect a watermarked tests, but that will need a lot more code